pkg/restore/merge_service_account.go

```diff
@@ -47,8 +47,8 @@ func mergeServiceAccounts(fromCluster, fromBackup *unstructured.Unstructured) (*
 
 	for i := len(backupSA.Secrets) - 1; i >= 0; i-- {
 		secret := &backupSA.Secrets[i]
-		if strings.HasPrefix(secret.Name, "default-token-") {
-			// Copy all secrets *except* default-token
+		if strings.HasPrefix(secret.Name, backupSA.ObjectMeta.Name+"-token-") {
```

Comment:
nit, but you can just say `backupSA.Name` since `ObjectMeta` is embedded.

Reply:
The nit has been picked.